ironfish/src/rpc/clients/ipcClient.ts

```diff
@@ -1,157 +1,12 @@
 /* This Source Code Form is subject to the terms of the Mozilla Public
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
-import { IPC, IpcClient } from 'node-ipc'
-import { Assert } from '../../assert'
-import { Event } from '../../event'
-import { createRootLogger, Logger } from '../../logger'
-import { ErrorUtils } from '../../utils'
-import { SocketRpcRequest } from '../adapters/socketAdapter/protocol'
-import { RpcConnectionRefusedError } from './errors'
-import { RpcClientConnectionInfo, RpcSocketClient } from './socketClient'
 
-const CONNECT_RETRY_MS = 2000
+import { createRootLogger, Logger } from '../../logger'
+import { RpcSocketClient } from './socketClient'
 
 export class RpcIpcClient extends RpcSocketClient {
-  ipc: IPC | null = null
-  ipcPath: string | null = null
-  client: IpcClient | null = null
-  isConnected = false
-  connection: Partial<RpcClientConnectionInfo>
-  retryConnect: boolean
-
-  onError = new Event<[error: unknown]>()
-
-  constructor(
-    connection: Partial<RpcClientConnectionInfo> = {},
-    logger: Logger = createRootLogger(),
-    retryConnect = false,
-  ) {
-    super(logger.withTag('ipcclient'))
-    this.connection = connection
-    this.retryConnect = retryConnect
-  }
-
-  async connect(options?: {
-    retryConnect?: boolean
-    connection?: Partial<RpcClientConnectionInfo>
-  }): Promise<void> {
-    const retryConnect = options?.retryConnect ?? this.retryConnect
-    const connection = { ...options?.connection, ...this.connection }
-
-    if (connection.mode === 'ipc' && !connection.socketPath) {
-      throw new Error('No IPC socket path given to connect to.')
-    }
-
-    if (connection.mode === 'tcp' && (!connection.host || !connection.port)) {
-      throw new Error('No IPC socket path given to connect to.')
-    }
-
-    const { IPC } = await import('node-ipc')
-    const ipc = new IPC()
```

Comment:
We should test this on Windows, if you haven't already -- looks like there may be some weirdness around how the IPC paths need to be specified that node-ipc was abstracting over https://nodejs.org/api/net.html#identifying-paths-for-ipc-connections

Reply:
Fixed.